borg/archiver.py

```diff
@@ -1416,20 +1416,21 @@ def run(self, args):
         setup_logging(level=args.log_level, is_serve=args.func == self.do_serve)  # do not use loggers before this!
         check_extension_modules()
         keys_dir = get_keys_dir()
-        if not os.path.exists(keys_dir):
-            os.makedirs(keys_dir)
-            os.chmod(keys_dir, stat.S_IRWXU)
         cache_dir = get_cache_dir()
-        if not os.path.exists(cache_dir):
-            os.makedirs(cache_dir)
-            os.chmod(cache_dir, stat.S_IRWXU)
-            with open(os.path.join(cache_dir, 'CACHEDIR.TAG'), 'w') as fd:
-                fd.write(textwrap.dedent("""
-                    Signature: 8a477f597d28d172789f06886806bc55
-                    # This file is a cache directory tag created by Borg.
-                    # For information about cache directory tags, see:
-                    #       http://www.brynosaurus.com/cachedir/
-                    """).lstrip())
+        if args.func != self.do_serve:  # remote serve does not need cache nor keys directories
+            if not os.path.exists(keys_dir):
+                os.makedirs(keys_dir)
+                os.chmod(keys_dir, stat.S_IRWXU)
+            if not os.path.exists(cache_dir):
+                os.makedirs(cache_dir)
+                os.chmod(cache_dir, stat.S_IRWXU)
+                with open(os.path.join(cache_dir, 'CACHEDIR.TAG'), 'w') as fd:
+                    fd.write(textwrap.dedent("""
+                        Signature: 8a477f597d28d172789f06886806bc55
+                        # This file is a cache directory tag created by Borg.
+                        # For information about cache directory tags, see:
+                        #       http://www.brynosaurus.com/cachedir/
+                        """).lstrip())
```

Comment:
well, guess it would work that way.

maybe a bit prettier (and having less code in archiver.py) would be to move the code that creates/inits the dirs into get_cache_dir and get_keys_dir. and remove the calls to get_cache_dir and get_keys_dir from here as the result is not needed here anyway.